src/rem/NumberLine.tsx

```diff
@@ -0,0 +1,134 @@
+import clsx from 'clsx';
+import { FC, ReactNode } from 'react';
+
+const Triangle: FC<{ up: boolean }> = ({ up }) =>
+  up ? (
+    <span className="text-red-500">{'\u25b2'}&nbsp;</span>
+  ) : (
+    <span className="text-green-500">{'\u25bc'}&nbsp;</span>
+  );
+
+export const NumberLine: FC<{
+  left: number;
+  middle: number;
+  right: number;
+
+  zeroLabel: string;
+  renderEnds: (n: number) => ReactNode;
+  renderMiddle: (n: number) => ReactNode;
+}> = ({ left, middle, right, zeroLabel, renderEnds, renderMiddle }) => {
+  const middleBubble = (
+    <div className="mx-auto bg-purple-100 rounded-lg px-4 py-1 text-lg leading-8">
+      <Triangle up={middle > 0} />
+      {renderMiddle(middle)}
+    </div>
+  );
+
+  // A triangle with a right angle at the bottom
+  const bubbleArrow = (
+    <div className="mx-auto text-purple-100">
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        width="16"
+        height="8"
+        viewBox="0 0 16 8"
+      >
+        <path d="M0 0 h 16 L 8 8 Z" fill="currentColor" />
+      </svg>
+    </div>
+  );
+
+  // There are 5 cases for the bar:
+  //
+  // 1. All negative: full width green
+  // 2. p20 positive, median and p80 negative: 1/4 red, zero mark, 3/4 green
+  // 3. p20 positive, median zero, p80 negative: 1/2 red, no mark, 1/2 green
+  // 4. p20 and median positive, p80 negative: 3/4 red, zero mark, 1/4 green
+  // 5. All positive: full width red
+  //
+  // This, plus the labels below it, are implemented with a 3-column 2-row grid.
+
+  const gridColClasses =
+    (left <= 0 && right <= 0) || (left > 0 && right > 0)
+      ? // p20 and p80 are the same sign: suppress zero mark
+        'grid-cols-[auto_0_auto]'
+      : middle < 0
+      ? // Mixed case, negative median: zero mark on left
+        'gap-x-[0.185rem] grid-cols-[25%_0.125rem_auto]'
+      : middle === 0
+      ? // Mixed case, zero median: suppress zero mark
+        'grid-cols-[auto_0_auto]'
+      : // Mixed case, positive median: zero mark on right
+        'gap-x-[0.185rem] grid-cols-[auto_0.125rem_25%]';
+
+  const lineAndLabels = (
+    <div
+      className={clsx(
+        'grid grid-rows-[0.75rem_auto] relative mt-1',
+        gridColClasses,
+      )}
+    >
+      <div
+        className={clsx(
+          'h-1 my-auto rounded-sm',
+          left <= 0 ? 'bg-green-500' : 'bg-red-500',
+        )}
+      ></div>
+      <div className="h-[0.625rem] my-auto bg-grey-400"></div>
+      <div
+        className={clsx(
+          'h-1 my-auto rounded-sm',
+          right <= 0 ? 'bg-green-500' : 'bg-red-500',
+        )}
+      ></div>
```

Comment:
The line and zero mark are drawn with divs because it's just easier than making an SVG scale the way I want it to.